### PR TITLE
#11: Improved path handling in buckets.

### LINKED ISSUE #11: IllegalArgumentException when using `listContents` method
## How to reproduce
```
        Bucket bucket = container.getDefaultBucket();
        Path pathToJar = Path.of("target/row-level-security-0.1.0-all-dependencies.jar");
        bucket.uploadFile(pathToJar, "jars/");
        System.out.println(bucket.listContents("/jars/"));
```

## Error message

```
java.lang.IllegalArgumentException: duplicate element:   |b  ՝AN            

	at java.base/java.util.ImmutableCollections$SetN.<init>(ImmutableCollections.java:587)
	at java.base/java.util.Set.of(Set.java:701)
	at com.exasol.bucketfs.Bucket.listContents(Bucket.java:86)
	at com.exasol.adapter.dialects.rls.RowLevelSecurityQueryRewriterIT.beforeAll(RowLevelSecurityQueryRewriterIT.java:33)
```

### PR DESCRIPTION
**Pull request by redcatbear**
Closes #11.

### CODE REVIEW

File: doc/user_guide/user_guide.md [hunk 1]
```diff
@@ -62,6 +62,14 @@ This is why **you should not use test containers in a production environment**.
 
 The Exasol test container provides access to buckets in BucketFS. This is useful if your tests need to work with files in buckets. If you for example want to test a UDF script, you can upload it prior to the test using a `Bucket` control object. 
 
+### Understanding Bucket Contents
+
+One thing you need to know about buckets objects inside a bucket are not stored in a hierarchical directory structure. Instead they are flat files that can have slashes in their names so that it looks like a directory structure at first glance.
```
Comment (morazow) [inline doc/user_guide/user_guide.md]: Suggestion:

`... objects inside a bucket is that they are not stored in a hierarchical directory structure.`

File: src/test/java/com/exasol/bucketfs/BucketTest.java [hunk 2]
```diff
@@ -36,17 +36,39 @@ void testGetDefaultBucket() {
...
     void testListBucketContents() throws BucketAccessException, InterruptedException {
-        assertThat(container.getDefaultBucket().listContents("/"), hasItem(startsWith("EXAClusterOS")));
+        assertThat(container.getDefaultBucket().listContents(), hasItem(startsWith("EXAClusterOS")));
+    }
+
+    // [itest->dsn~bucket-lists-its-contents~1]
+    @Test
+    void testListBucketContentsWithRootPath() throws BucketAccessException, InterruptedException {
+        assertThat(container.getDefaultBucket().listContents(), hasItem(startsWith("EXAClusterOS")));
     }
```
Comment (morazow) [inline src/test/java/com/exasol/bucketfs/BucketTest.java]: These are testing same functionality. Did you intended the first one include another path?

File: doc/user_guide/user_guide.md (hunk 1 above)
Comment (redcatbear) [reply doc/user_guide/user_guide.md]: Fixed.

File: src/test/java/com/exasol/bucketfs/BucketTest.java (hunk 2 above)
Comment (redcatbear) [reply src/test/java/com/exasol/bucketfs/BucketTest.java]: Fixed.

**Pull request merged**